src/server_manager/cloud/digitalocean_api.ts

```diff
@@ -124,16 +124,44 @@ class RestApiSession implements DigitalOceanSession {
     // confusing email with their droplet password, which could get mistaken for
     // an invite.
     return this.registerKey_(dropletName, publicKeyForSSH).then((keyId: number) => {
-      return this.request<{droplet: DropletInfo}>('POST', 'droplets', {
-        name: dropletName,
-        region,
-        size: dropletSpec.size,
-        image: dropletSpec.image,
-        ssh_keys: [keyId],
-        user_data: dropletSpec.installCommand,
-        tags: dropletSpec.tags,
-        ipv6: true,
-      });
+      return this.makeCreateDropletRequest(dropletName, region, keyId, dropletSpec);
+    });
+  }
+
+  private makeCreateDropletRequest(
+      dropletName: string, region: string, keyId: number,
+      dropletSpec: DigitalOceanDropletSpecification): Promise<{droplet: DropletInfo}> {
+    let requestCount = 0;
+    const MAX_REQUESTS = 10;
+    return new Promise((fulfill, reject) => {
+      const makeRequestRecursive = () => {
+        ++requestCount;
+        this.request<{droplet: DropletInfo}>('POST', 'droplets', {
+              name: dropletName,
+              region,
+              size: dropletSpec.size,
+              image: dropletSpec.image,
+              ssh_keys: [keyId],
+              user_data: dropletSpec.installCommand,
+              tags: dropletSpec.tags,
+              ipv6: true,
+            })
+            .then(fulfill)
+            .catch((e) => {
+              const ACCOUNT_NOT_YET_FINALIZED_ERROR =
+                  'We\'re finalizing your account setup. Please try again in 30 seconds.';
+              if (e.message === ACCOUNT_NOT_YET_FINALIZED_ERROR && requestCount < MAX_REQUESTS) {
+                // DigitalOcean is still validating this account and may take
+                // up to 30 seconds.  We can retry more frequently to see when
+                // this error goes away.
+                const RETRY_TIMEOUT_MS = 5000;
```

Comment:
You could declare this at a higher scope, e.g. near `MAX_REQUESTS`.

Reply:
Done